src/fireedge/src/modules/components/List/NestedLabelTree.js

```diff
@@ -0,0 +1,752 @@
+    ;(successAddLabel || successGroupAddLabel) &&
+      enqueueSuccess(`${T.AddedNewLabel}!`)
+  }, [successAddLabel, successGroupAddLabel])
+
+  const {
+    user: { ID: uId } = {},
+    groups: ownGroups,
+    isOneAdmin = false,
+    labels: fetchedLabels = {},
+    defaultLabels = {},
+  } = useAuth()
+
+  const groups = [...new Set(ownGroups.concat(isOneAdmin ? allGroups : []))]
+
+  const groupIdMap = useMemo(
+    () =>
+      groups?.reduce((acc, group) => {
+        acc[group?.NAME] = group?.ID
+
+        return acc
+      }, {}),
+    [groups]
+  )
+
+  const fmtDefaultLabels = useMemo(
+    () =>
+      Object.fromEntries(
+        Object.entries(defaultLabels).map(([type, paths]) => {
+          const root = {}
+
+          const validPaths = paths.filter((path) => {
+            if (type !== 'group') return true
+
+            const segments = path.split('/').filter(Boolean)
+
+            return (
+              segments.length === 0 || Object.hasOwn(groupIdMap, segments?.[0])
+            )
+          })
+
+          for (const path of validPaths) {
+            path
+              .split('/')
+              .filter(Boolean)
+              .reduce((acc, segment, idx) => {
+                const shouldPrefix =
+                  (type === 'group' && idx > 0) || type === 'user'
+
+                const key = shouldPrefix ? `$${segment}` : segment
+
+                acc[key] ??= {}
+
+                return acc[key]
+              }, root)
+          }
+
+          return [type, root]
+        })
+      ),
+    [defaultLabels, groupIdMap]
+  )
+
+  const parsedLabels = useMemo(
+    () => ({
+      user: merge(
+        {},
+        get(fmtDefaultLabels, 'user', {}),
+        get(fetchedLabels, 'user', {})
+      ),
+      group: merge(
+        {},
+        get(fmtDefaultLabels, 'group', {}),
+        get(fetchedLabels, 'group', {})
+      ),
+    }),
+    [fetchedLabels, fmtDefaultLabels]
+  )
+
+  const editableGroups = useMemo(
+    () =>
+      groups?.reduce((acc, group) => {
+        const groupName = group?.NAME
+        const groupAdmins = [].concat(group?.ADMINS).filter(Boolean)
+
+        if (!groupName) return acc
+
+        if (isOneAdmin) {
+          acc[groupName] = true
+        } else {
+          if (!groupAdmins) return acc
+          const adminIDs =
+            groupAdmins?.map(({ ID }) => ID).filter(Boolean) ?? []
+          acc[groupName] = adminIDs.includes(uId)
+        }
+
+        return acc
+      }, {}),
+    [uId, groups]
+  )
+
+  const handleOpenAddModal = (parentNodeId = null, nodeType = 'user') => {
+    setAddLabelParentNodeId(parentNodeId)
+    setAddLabelType(nodeType)
+    setAddLabelModalOpen(true)
+  }
+
+  const handleRemoveLabel = async (nodeId, type) => {
+    const [label, id] = formatLabel(nodeId, type)
+
+    await apiRemoveLookup?.[type]?.({ id, label })
+  }
+
+  const formatLabel = (label, type) => {
+    const [splitPath, id] = partition(label.split('/'), (l) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).